pkg/controller/metrics/metricscontroller.go

```diff
@@ -183,14 +199,16 @@ func (c *controller) GetInstantMetrics(opt *pb.GetMetricsOpts) (*[]model.MetricS
 		if err != nil {
 			log.Error(err)
 		}
+		return metrics, err
 
 	}
-	return &metrics, err
+	//no response
+	return nil, err
 }
 
-func (c *controller) GetRangeMetrics(opt *pb.GetMetricsOpts) (*[]model.MetricSpec, error) {
+func (c *controller) GetRangeMetrics(opt *pb.GetMetricsOpts) ([]*model.MetricSpec, error) {
 
-	var metrics []model.MetricSpec
+	//var metrics []model.MetricSpec
 	// make a call to Prometheus, convert the response to our format, return
 	response, err := http.Get("http://localhost:9090/api/v1/query_range?query=" + opt.MetricName + "&start=" + opt.StartTime + "&end=" + opt.EndTime + "&step=30")
```

Comment:
Should be configurable.

Reply:
https://github.com/opensds/opensds/issues/691